osu.Game.Rulesets.Osu/Difficulty/Skills/Speed.cs

```diff
@@ -20,20 +22,119 @@ public class Speed : Skill
         private const double pi_over_4 = Math.PI / 4;
         private const double pi_over_2 = Math.PI / 2;
 
+        /// <summary>
+        /// streamaimconst is derived from the last stream of Megalovania [Excors] 9* map (Unit:pixel/ms)
+        /// </summary>
+        private const double streamaimconst = 2.42;
+
+        /// <summary>
+        /// splitter value for jumpDistances2 list
+        /// </summary>
+        private int sdsplitcounter;
+
+        /// <summary>
+        /// splitter value for jumpDistances list
+        /// </summary>
+        private int sdsplitcounter2;
+
+        /// <summary>
+        /// Final result of SD calculation and scaling added at the result of StrainValueOf(DifficultyHitObject current)
+        /// </summary>
+        private double sdstrainmult;
+
+        /// <summary>
+        /// jumpDistances stores acceleration as a list
+        /// </summary>
+        private readonly List<double> jumpDistances = new List<double>();
+
+        /// <summary>
+        /// strainTimes store StrainTime(s) as a list
+        /// </summary>
+        private readonly List<double> strainTimes = new List<double>();
+
+        /// <summary>
+        /// jumpDistances2 store sectionvelocity as a list.
+        /// </summary>
+        private readonly List<double> jumpDistances2 = new List<double>();
```

Comment:
This xmldoc both restates the variable name, is entirely unreadable, and does not explain the simultaneous existence of `jumpDistances` and `jumpDistances2`.

Reply:
Do you want me to change the names?

Reply:
That'd be a good first step, yes. Having names of the sort `variable` and `variable2` is one of the very basic examples of bad programming practice. Variable naming should be as descriptive as possible.